salt/utils/win_lgpo_netsh.py

```diff
@@ -195,51 +189,61 @@ def get_settings(profile, section, store="local"):
         raise ValueError(f"Incorrect section: {section}")
     if store.lower() not in ("local", "lgpo"):
         raise ValueError(f"Incorrect store: {store}")
-    command = f"show {profile}profile {section}"
-    # run it
-    results = _netsh_command(command=command, store=store)
-    # sample output:
-    # Domain Profile Settings:
-    # ----------------------------------------------------------------------
-    # LocalFirewallRules                    N/A (GPO-store only)
-    # LocalConSecRules                      N/A (GPO-store only)
-    # InboundUserNotification               Disable
-    # RemoteManagement                      Disable
-    # UnicastResponseToMulticast            Enable
-
-    # if it's less than 3 lines it failed
-    if len(results) < 3:
-        raise CommandExecutionError(f"Invalid results: {results}")
-    ret = {}
-    # Skip the first 2 lines. Add everything else to a dictionary
-    for line in results[3:]:
-        ret.update(dict(list(zip(*[iter(re.split(r"\s{2,}", line))] * 2))))
-
-    # Remove spaces from the values so that `Not Configured` is detected
-    # correctly
-    for item in ret:
-        ret[item] = ret[item].replace(" ", "")
-
-    # special handling for firewallpolicy
-    if section == "firewallpolicy":
-        inbound, outbound = ret["Firewall Policy"].split(",")
-        return {"Inbound": inbound, "Outbound": outbound}
-
-    return ret
 
+    # Build the powershell command
+    cmd = ["Get-NetFirewallProfile"]
+    if profile:
+        cmd.append(profile)
+    if store and store.lower() == "lgpo":
+        cmd.extend(["-PolicyStore", "localhost"])
 
-def get_all_settings(profile, store="local"):
-    """
-    Gets all the properties for the specified profile in the specified store
+    # Run the command
+    settings = salt.utils.win_pwsh.run_dict(cmd)
+
+    # A successful run should return a dictionary
+    if not settings:
+        raise CommandExecutionError("LGPO NETSH: An unknown error occurred")
+
+    # Remove the junk
+    for setting in list(settings.keys()):
```

Comment:
```suggestion
    for setting in list(settings):
```
This is enough, not blocking merge because of it though.